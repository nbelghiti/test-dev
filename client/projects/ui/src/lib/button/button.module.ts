import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';

import { ButtonComponent } from './button/button.component';
import { IconButtonComponent } from './icon-button/icon-button.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
	],
	declarations: [
		ButtonComponent,
		IconButtonComponent,
	],
	exports: [
		ButtonComponent,
		IconButtonComponent,
	],
})
export class ButtonModule {}
