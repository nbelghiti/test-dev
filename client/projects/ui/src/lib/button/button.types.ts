export enum ButtonType {
	primary = '',
	secondary = 'c-button--secondary',
	tertiary = 'c-button--tertiary',
	destructive = 'c-button--destructive',
}
