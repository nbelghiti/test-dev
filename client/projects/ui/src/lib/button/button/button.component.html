<button
	*ngIf="!link; else isLink"
	class="c-button {{ type }} {{ classes }}"
	[title]="buttonTitle"
	[ngClass]="{
		'c-button--disabled': disabled,
		'c-button--loading': loading,
		'c-button--fluid': fluid
	}"
	[disabled]="disabled"
	(click)="loading ? null : clicked.emit($event)"
>
	<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
</button>

<ng-template #isLink>
	<a
		class="c-button {{ type }} {{ classes }}"
		[ngClass]="{
			'c-button--disabled': disabled,
			'c-button--loading': loading,
			'c-button--fluid': fluid
		}"
		[routerLink]="!disabled && link"
	>
		<ng-container [ngTemplateOutlet]="buttonContent"></ng-container>
	</a>
</ng-template>

<ng-template #buttonContent>
	<div
		*ngIf="iconBefore"
		class="c-button__icon c-button__icon--before"
	>
		<ui-icon
			[name]="iconBefore"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>

	<span class="c-button__text">
		<ng-content></ng-content>
	</span>

	<i class="c-button__loader fas fa-spinner fa-spin"></i>

	<div
		*ngIf="iconAfter"
		class="c-button__icon c-button__icon--after"
	>
		<ui-icon
			[name]="iconAfter"
			[size]="type === 'c-button--tertiary' && 'small'"
		>
		</ui-icon>
	</div>
</ng-template>
