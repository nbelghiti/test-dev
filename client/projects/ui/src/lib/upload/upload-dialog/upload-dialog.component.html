<div class="c-dialog__content">
	<ui-heading
		[level]="HeadingLevel.heading2"
		[title]="'UPLOAD__SUBTITLE' | translate"
		classes="c-dialog__title"
	>
	</ui-heading>

	<ui-upload-dropzone
		[data]="data"
		[files]="files"
		[uploading]="data?.uploading$ | async"
		[uploaded]="uploaded"
	></ui-upload-dropzone>

	<div *ngIf="response?.errors" class="u-margin-top-3x">
		<div class="u-margin-bottom-1x">
			<ui-heading
				[level]="HeadingLevel.heading3"
				[title]="response?.message"
			>
			</ui-heading>
		</div>

		<ui-list class="c-dialog__list">
			<ng-container *ngFor="let error of response?.errors">
				<ui-info-item
					*uiListItem
						[text]="error"
						[icon]="IconName.exclamationCircle"
					>
					</ui-info-item>
			</ng-container>
		</ui-list>
	</div>
</div>

<div *ngIf="!uploaded; else closeButton" class="c-dialog__actions">
	<div class="c-dialog__action-item">
		<ui-button
			(click)="onCancel()"
			[type]="ButtonType.secondary"
			[disabled]="data?.uploading$ | async"
		>
			{{ 'GENERAL__CANCEL' | translate }}
		</ui-button>
	</div>

	<div class="c-dialog__action-item">
		<ui-button
			(click)="onUpload()"
			[disabled]="!files.size || (data?.uploading$ | async)"
		>
			{{ 'UPLOAD__TITLE' | translate }}
		</ui-button>
	</div>
</div>

<ng-template #closeButton>
	<div class="c-dialog__actions">
		<div class="c-dialog__action-item">
			<ui-button
				(click)="onCancel()"
			>
				{{ 'UPLOAD__VIEW_CHANGES' | translate }}
			</ui-button>
		</div>
	</div>
</ng-template>
