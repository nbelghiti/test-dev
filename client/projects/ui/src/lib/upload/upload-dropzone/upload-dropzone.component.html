<div
	*ngIf="!uploaded; else isUploaded"
	class="c-upload-dropzone"
>
	<ui-loader
		[simpleLoader]="true"
		[uploadingMsg]="true"
		[loading]="data?.uploading$ | async"
	>
		<div
			for="file"
			class="c-upload-dropzone__content"
			[class.is-disabled]="uploading"
		>
			<input
				class="c-upload-dropzone__input" type="file"
				(change)="onFilesAdded()"
				[disabled]="uploading"
				#file
			/>

			<div class="c-upload-dropzone__label">
				<ui-icon [name]="IconName.plus"></ui-icon>

				<span class="c-upload-dropzone__label-text">
					{{ 'UPLOAD__INSTRUCTIONS' | translate }}
				</span>
			</div>
		</div>
	</ui-loader>

	<ul class="c-upload-dropzone__list">
		<li
			*ngFor="let file of files"
			class="c-upload-dropzone__list-item"
		>
			<ui-icon [name]="IconName.file"></ui-icon>

			{{ file.name }}
		</li>
	</ul>
</div>

<ng-template #isUploaded>
	<div class="c-upload-dropzone__label">
		<ui-icon [name]="IconName.check"></ui-icon>

		<span class="c-upload-dropzone__success-message">
			{{ 'UPLOAD__DONE' | translate }}
		</span>
	</div>

	<div class="u-margin-top-2x">
		<small>
			{{ 'UPLOAD__HINT' | translate }}
		</small>
	</div>
</ng-template>
