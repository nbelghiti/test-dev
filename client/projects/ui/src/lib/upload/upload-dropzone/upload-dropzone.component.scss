// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Upload Dropzone
// =============================================================================
// .c-upload-dropzone {}

// =============================================================================
// :: Elements
// =============================================================================
.c-upload-dropzone__input {
	position: absolute;
	z-index: 9;

	width: 100%;
	height: 100%;
	overflow: hidden;

	opacity: 0;

	cursor: pointer;
}

.c-upload-dropzone__content {
	position: relative;

	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	width: 100%;
	height: 100%;
	min-height: 14rem;

	text-align: center;

	border: 0.1rem dashed $border-color-grey-76;
	background-color: $background-color-grey-96;
	color: $text-color-grey-46;

	outline-offset: 0.2rem;

	&.is-disabled input {
		cursor: wait;
	}

	&:not(.is-disabled) {

		&:focus,
		&:hover {
			outline: $border-width-base solid $primary-color;
		}
	}
}

.c-upload-dropzone__label {
	display: flex;
	justify-content: center;
	align-items: center;
}

.c-upload-dropzone__label-text {
	margin-left: 1rem;

	text-decoration: underline;
}

.c-upload-dropzone__success-message {
	margin-left: 1rem;
}

.c-upload-dropzone__list {
	margin-top: 1.6rem;
}

.c-upload-dropzone__list-item {
	margin: 0.8rem 0;

	&:first-child {
		margin-top: 0;
	}

	&:last-child {
		margin-bottom: 0;
	}
}
