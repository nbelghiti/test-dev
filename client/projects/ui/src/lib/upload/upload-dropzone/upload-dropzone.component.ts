import { Component, Input, ViewChild } from '@angular/core';

import { IDialogData } from '../../dialog/dialog.types';
import { IconName } from '../../icon/icon.types';

@Component({
	selector: 'ui-upload-dropzone',
	templateUrl: './upload-dropzone.component.html',
	styleUrls: [ './upload-dropzone.component.scss' ],
})
export class UploadDropzoneComponent {
	@Input() public data: IDialogData;
	@Input() public uploading: boolean = false;
	@Input() public uploaded: boolean = false;
	@Input() public files: Set<File> = new Set<File>();
	@ViewChild('file', { static: false }) public file;

	public IconName: typeof IconName = IconName;

	public onFilesAdded(): void {
		const uploadedFiles = this.file.nativeElement.files;

		if (this.data && this.data.multiple) {
			for (let i = 0; i < uploadedFiles.length; i++) {
				this.files.add(uploadedFiles[i]);
			}
		}

		this.files.clear();

		this.files.add(uploadedFiles[0]);
	}
}
