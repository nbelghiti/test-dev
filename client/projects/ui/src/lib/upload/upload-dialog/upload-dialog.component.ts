import { Component, HostBinding, OnDestroy } from '@angular/core';
import { isNil } from 'ramda';
import { Subject } from 'rxjs';
import { filter, takeUntil } from 'rxjs/operators';

import { ButtonType } from '../../button/button.types';
import { IDialog, IDialogActions, IDialogData } from '../../dialog/dialog.types';
import { HeadingLevel } from '../../heading/heading.types';
import { IconName } from '../../icon/icon.types';
import { IUploadResponse } from '../upload.types';

@Component({
	selector: 'ui-upload-dialog',
	templateUrl: './upload-dialog.component.html',
})
export class UploadDialogComponent implements IDialog, OnDestroy {
	@HostBinding('class.c-dialog') public dialogClassName = true;

	public IconName: typeof IconName = IconName;
	public uploaded: boolean = false;
	public response: IUploadResponse;
	public actions: IDialogActions;
	public data: IDialogData;
	public files: Set<File> = new Set<File>();
	public HeadingLevel = HeadingLevel;
	public ButtonType = ButtonType;

	private destroyed$: Subject<boolean> = new Subject<boolean>();

	public closeDialog(): void {
		if (this.actions.close) {
			this.actions.close();
		}
	}

	public onCancel(): void {
		this.closeDialog();
	}

	public onUpload(): void {
		if (!this.files.size) {
			return;
		}

		const files = Array.from(this.files);

		this.data.uploading$
			.pipe(
				takeUntil(this.destroyed$),
				filter((uploading: boolean) => !uploading)
			)
			.subscribe(() => this.uploaded = true);

		this.data.response$
			.pipe(
				takeUntil(this.destroyed$),
				filter((response: IUploadResponse) => !isNil(response))
			)
			.subscribe((response: IUploadResponse) => this.response = response);

		this.actions.upload(
			this.data.multiple ? files : files[0]
		);
	}

	public ngOnDestroy(): void {
		this.destroyed$.next(true);
		this.destroyed$.complete();
	}
}
