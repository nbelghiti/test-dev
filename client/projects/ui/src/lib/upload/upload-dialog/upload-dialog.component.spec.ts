
import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';
import { TranslateModule } from '@ngx-translate/core';

import { ButtonComponent } from '../../button/button/button.component';
import { DialogModule } from '../../dialog/dialog.module';
import { HeadingComponent } from '../../heading/heading.component';
import { IconComponent } from '../../icon/icon.component';
import { InfoItemComponent } from '../../info-item/info-item.component';
import { LinkComponent } from '../../link/link.component';
import { ListComponent } from '../../list/list.component';
import { LoaderComponent } from '../../loader/loader.component';
import { UploadDropzoneComponent } from '../upload-dropzone/upload-dropzone.component';

import { UploadDialogComponent } from './upload-dialog.component';

describe('UploadDialogComponent', () => {
	let component: UploadDialogComponent;
	let fixture: ComponentFixture<UploadDialogComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				TranslateModule.forRoot(),
				FormsModule,
				DialogModule
			],
			declarations: [
				UploadDialogComponent,
				IconComponent,
				LinkComponent,
				ButtonComponent,
				UploadDropzoneComponent,
				LoaderComponent,
				HeadingComponent,
				InfoItemComponent,
				ListComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(UploadDialogComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
