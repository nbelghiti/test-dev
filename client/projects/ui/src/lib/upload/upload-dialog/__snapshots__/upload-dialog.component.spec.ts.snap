// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadDialogComponent Should render correctly 1`] = `
<ui-upload-dialog
  ButtonType={[Function Object]}
  HeadingLevel={[Function Object]}
  IconName={[Function Object]}
  destroyed$={[Function Subject]}
  dialogClassName={[Function Boolean]}
  files={[Function Set]}
  uploaded="false"
>
  <div
    class="c-dialog__content"
  >
    <ui-heading
      classes="c-dialog__title"
      ng-reflect-classes="c-dialog__title"
      ng-reflect-level="2"
      ng-reflect-title="UPLOAD__SUBTITLE"
    >
      <div
        class="c-heading c-dialog__title"
        ng-reflect-klass="c-heading c-dialog__title"
        ng-reflect-ng-class="[object Object]"
      >
        
        
        
        <h2>
          UPLOAD__SUBTITLE
        </h2>
        
        <div
          class="c-heading__meta"
        >
          
          
          
        </div>
      </div>
    </ui-heading>
    <ui-upload-dropzone
      ng-reflect-files="[object Set]"
      ng-reflect-uploaded="false"
    >
      
      <div
        class="c-upload-dropzone"
      >
        <ui-loader
          ng-reflect-simple-loader="true"
          ng-reflect-uploading-msg="true"
        >
          
          <div
            class="c-upload-dropzone__content"
            for="file"
          >
            <input
              class="c-upload-dropzone__input"
              type="file"
            />
            <div
              class="c-upload-dropzone__label"
            >
              <ui-icon
                class="c-icon"
                ng-reflect-name="plus"
              >
                <i
                  class="c-icon__glyph   fas fa-plus"
                />
              </ui-icon>
              <span
                class="c-upload-dropzone__label-text"
              >
                 UPLOAD__INSTRUCTIONS 
              </span>
            </div>
          </div>
          
          
        </ui-loader>
        <ul
          class="c-upload-dropzone__list"
        >
          
        </ul>
      </div>
      
    </ui-upload-dropzone>
    
  </div><div
    class="c-dialog__actions"
  >
    <div
      class="c-dialog__action-item"
    >
      <ui-button
        ng-reflect-type="c-button--secondary"
      >
        
        <button
          class="c-button c-button--secondary"
          ng-reflect-klass="c-button c-button--secondary "
          ng-reflect-ng-class="[object Object]"
          title=""
        >
          
          
          <span
            class="c-button__text"
          >
             GENERAL__CANCEL 
          </span>
          <i
            class="c-button__loader fas fa-spinner fa-spin"
          />
          
          
        </button>
        
        
      </ui-button>
    </div>
    <div
      class="c-dialog__action-item"
    >
      <ui-button
        ng-reflect-disabled="true"
      >
        
        <button
          class="c-button c-button--disabled"
          disabled=""
          ng-reflect-klass="c-button  "
          ng-reflect-ng-class="[object Object]"
          title=""
        >
          
          
          <span
            class="c-button__text"
          >
             UPLOAD__TITLE 
          </span>
          <i
            class="c-button__loader fas fa-spinner fa-spin"
          />
          
          
        </button>
        
        
      </ui-button>
    </div>
  </div>
</ui-upload-dialog>
`;
